packages/fluentui/docs/src/examples/components/RadioGroup/Usage/RadioGroupCustomExample.shorthand.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { Divider, RadioGroup, Input } from '@fluentui/react-northstar';
+import { useBooleanKnob } from '@fluentui/docs-components';
+
+const RadioGroupCustomExample = props => {
+  const [vertical] = useBooleanKnob({
+    name: 'vertical',
+    initialValue: false,
+  });
+
+  const [state, dispatch] = React.useReducer(
+    (state, action) => {
+      switch (action.type) {
+        case 'UPDATE':
+          return { ...state, ...action.payload };
+        case 'UPDATE_VALUE':
+          return { ...state, selectedValue: action.selectedValue };
+        default:
+          return state;
+      }
+    },
+    {
+      selectedValue: 'capricciosa',
+      inputTabIndex: '-1',
+    },
+  );
+
+  const getItems = () => {
+    return [
+      { name: 'pizza', key: 'Capricciosa', label: 'Capricciosa', value: 'capricciosa' },
+      {
+        name: 'pizza',
+        key: 'Prosciutto',
+        label: 'Prosciutto',
+        value: 'prosciutto',
+        disabled: true,
+      },
+      {
+        name: 'pizza',
+        key: 'Custom',
+        label: 'Choose your own',
+        children: (Component, { key, ...props }) => {
+          return (
+            <div
+              key={key}
+              style={{
+                display: 'inline-flex',
+              }}
+            >
+              <Component {...props} />
+              <Input
+                onFocus={() => {
+                  dispatch({ type: 'UPDATE_VALUE', selectedValue: 'custom' });
+                }}
+                input={{ tabIndex: state.inputTabIndex }}
```

Comment:
We have tab index in the state on for this handle and we are changing it based in this condition on the `handleChange` no need to double check here

Reply:
But `tabIndex` depends only on `selectedValue`, in this case we can remove it from the state and use `React.useState()` to instead of `React.useReducer()`.